### PR TITLE
Remove the usage of must_cache and async_cache

### PR DESCRIPTION
**Pull request by Jackson-Wang-7**
### What changes are proposed in this pull request?
current version didn't support must_cache and async_cache, so remove the usage of must_cache and async_cache usage in code and mark deprecated to prevent any use.

### Why are the changes needed?
This change focuses on removing the write type of MUST_CACHE and ASYNC_CACHE, and https://github.com/Alluxio/alluxio/pull/17980 would remove the master-side code about async persist.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Comment by yyongycy** [comment]
Please also have a simple test to make sure it works.

File: dora/tests/src/test/java/alluxio/client/cli/fs/command/ChecksumCommandIntegrationTest.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public final class ChecksumCommandIntegrationTest extends AbstractFileSystemShel
    */
   @Test
   public void checksum() throws Exception {
-    FileSystemTestUtils.createByteFile(sFileSystem, "/testFile", WritePType.MUST_CACHE, 10);
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFile", WritePType.CACHE_THROUGH, 10);
```
Comment (jiacheliu3) [inline dora/tests/src/test/java/alluxio/client/cli/fs/command/ChecksumCommandIntegrationTest.java:39]: For this kind of manually set write types, what do you think about declaring a shared var?
```
class ChecksumCommandIntegrationTest {
  public static final WritePType DEFAULT_WRITE_TYPE = WritePType.CACHE_THROUGH;

  // keep using this constant throughout your test
}
```

You don't save any code in this PR, but you may save a lot in later PRs (if you tweak the default write type).
All these test class files may share the same `DEFAULT_WRITE_TYPE` if you find a good place to declare this constant.

Comment (Jackson-Wang-7) [reply dora/tests/src/test/java/alluxio/client/cli/fs/command/ChecksumCommandIntegrationTest.java:39]: @jiacheliu3 I'm not sure if we expect a default value for the different unit tests. For the general default value, we only need to set `DEFAULT_WRITE_TYPE` in the cluster resouce configuration in some based test class. 
Except that, I saw the most cases is it wants to excute a specific opration with a specific write type. we can not easily replace them with a `DEFAULT_WRITE_TYPE` in general.

File: dora/tests/src/test/java/alluxio/client/cli/fsadmin/command/ReportCommandIntegrationTest.java [hunk 2]
```diff
@@ -16,13 +16,15 @@
...
 import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
 /**
  * Tests for report command.
  */
 public final class ReportCommandIntegrationTest extends AbstractFsAdminShellTest {
   @Test
+  @Ignore
```
Comment (jiacheliu3) [inline dora/tests/src/test/java/alluxio/client/cli/fsadmin/command/ReportCommandIntegrationTest.java:27]: add `@DoraTestTodoItem`?

**Review by jiacheliu3 — APPROVED** [review]
LGTM with one minor comment, thanks for the work!

**Comment by Jackson-Wang-7** [comment]
alluxio-bot, merge this please

**Pull request merged**